File: 2021/09/20/main.go
package main

func tictactoe(moves [][]int) string {
	grid := [3][3]int{}
	for i, move := range moves {
		grid[move[0]][move[1]] = i&1 + 1
	}

	for i := 0; i < 3; i++ {
		if player := grid[i][i]; player != 0 {
			if (grid[i][0] == grid[i][1] && grid[i][0] == grid[i][2]) ||
				(grid[0][i] == grid[1][i] && grid[0][i] == grid[2][i]) {
				return string(byte('@' + player))
			}
		}
	}

	if player := grid[1][1]; player != 0 {
		if (grid[0][0] == grid[2][2] && grid[0][0] == player) ||
			(grid[2][0] == grid[0][2] && grid[2][0] == player) {
			return string(byte('@' + player))
		}
	}

	if len(moves) != 9 {
		return "Pending"
	}

	return "Draw"
}

File: 2021/09/20/main.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        a, b = 1, -1
        player = a
        rows, cols = [0] * 3, [0] * 3
        up, down = 0, 0

        for r, c in moves:
            rows[r] += player
            cols[c] += player
            if r == c:
                up += player
            if r + c == 2:
                down += player

            player = -player

        lines = rows + cols + [up, down]
        if any(line == 3 * a for line in lines):
            return "A"
        if any(line == 3 * b for line in lines):
            return "B"
        return "Draw" if len(moves) == 9 else "Pending"

